modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java

```diff
@@ -718,11 +718,10 @@ public SlotCoordinator getSlotCoordinator() {
      * Store retained messages in the message store.
      *
      * @see org.wso2.andes.kernel.AndesMessageMetadata#retain
-     * @param retainList
-     *         Retained message list
+     * @param retainMap Retained message Map
      */
-    public void storeRetainedMessages(List<AndesMessage> retainList) throws AndesException {
-        messageStore.storeRetainedMessages(retainList);
+    public void storeRetainedMessages(HashMap<String,AndesMessage> retainMap) throws AndesException {
```

Comment:
Shall we use  `Map<String,AndesMessage>` instead of `HashMap<String,AndesMessage>`